mllib/src/test/scala/org/apache/spark/ml/regression/DecisionTreeRegressorSuite.scala

```diff
@@ -73,6 +76,56 @@ class DecisionTreeRegressorSuite extends SparkFunSuite with MLlibTestSparkContex
     MLTestingUtils.checkCopy(model)
   }
 
+  test("training with weighted data") {
```

Comment:
Since the data generation process for this test is identical to the classifier test, perhaps we could reuse the code and put it in `TreeTests`? It's basically just copy pasted as is.